.circleci/config.yml

```diff
@@ -1,16 +1,69 @@
-version: 2
+version: 2.1
+
+orbs:
+  python: circleci/python@2.1.1
+
+executors:
+  app-executor:
+    docker:
+      - image: cimg/python:3.10
+        auth:
+          username: $DOCKER_USER
+          password: $DOCKER_PASS
+
+aliases:
+  - &python-executor
+      name: python/default
+      tag: "3.10"
 
 fast-checkout: &fast-checkout
   attach_workspace:
     at: /tmp/workspace
 
 jobs:
+  requirements:
+    executor: *python-executor
+    steps:
+      - checkout
+      - python/install-packages:
+          args: "--no-interaction --no-root"
+          pkg-manager: poetry
+      - run:
+          command: |
+            poetry check
+
+  black:
+    executor: *python-executor
+    steps:
+      - checkout
+      - python/install-packages:
+          args: "--no-interaction --no-root --only dev"
+          pkg-manager: poetry
+      - run:
+          name: List added, copied, modified, and renamed *py files
+          command: |
+            git diff --name-only --diff-filter=ACMR origin/main | grep -E "(.py$)" > diff.txt || true
+      - run:
+          name: Black code formatting
+          command: |
+            poetry run black --check --diff --color --config=pyproject.toml . &> lint_checks.txt || true
+      - run:
+          name: Diff-based black
+          command: |
+            grep -Ff diff.txt lint_checks.txt > lint_errors.txt || true
+            if [ -s lint_errors.txt ]; then
+              cat lint_errors.txt
+              printf 'Run the following command to fix your branch:\n make fixes'
+              exit 1
+            fi
+
   persist-checkout:
-    docker:
-      - image: circleci/python:3.9
-        auth:
-          username: $DOCKER_USER
-          password: $DOCKER_PASS
+    executor: app-executor
+    # docker:
+    #   - image: circleci/python:3.9
+    #     auth:
+    #       username: $DOCKER_USER
+    #       password: $DOCKER_PASS
```

Comment:
Can you remove this commented code? We can always rollback if needed

Reply:
Done